src/raftstore/store/snap.rs

```diff
@@ -509,7 +509,7 @@ mod v1 {
                     let cap = cmp::min(self.left, DEFAULT_READ_BUFFER_SIZE);
                     let mut buf = vec![0; cap];
                     while self.left > 0 {
-                        try!(self.read(&mut buf));
+                        try!(self.read_exact(&mut buf));
```

Comment:
Yep, and this will generate a warning
warning: this let-binding has unit value. Consider omitting `let _ =`
   --> src/raftstore/store/snap.rs:512:25
    |
512 |                         let _ = try!(self.read_exact(&mut buf));
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(let_unit_value)] on by default
    = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#let_unit_value